### PR TITLE
Synchronize player inventories

### PR DESCRIPTION
**Pull request by StephenCWills**
Synchronizes changes to the player's inventory to remote clients. There are definitely some missing edge cases, but this can get us started on resolving many of the sources of desync discussed in #4074.

Here is a video demonstration of how the synchronization looks when testing in conjunction with https://github.com/StephenCWills/devilutionX/commit/5df65ff54f6ee6fef6d13cb23e01d9710f4a0060:

https://user-images.githubusercontent.com/9203145/183315099-ff6d09f9-d43a-4304-bc2a-72db5a11a4b3.mp4

### CODE REVIEW

File: Source/msg.cpp [hunk 1]
```diff
@@ -2893,6 +2949,47 @@ void NetSendCmdDelItem(bool bHiPri, uint8_t bLoc)
...
+	int8_t invListIndex = abs(MyPlayer->InvGrid[invGridIndex]) - 1;
+	Item &item = MyPlayer->InvList[invListIndex];
+
+	cmd.bCmd = CMD_CHANGEINVITEMS;
+	cmd.bLoc = invGridIndex;
+	cmd.wIndx = item.IDidx;
+	cmd.wCI = item._iCreateInfo;
+	cmd.dwSeed = item._iSeed;
+	cmd.bId = item._iIdentified ? 1 : 0;
+	cmd.dwBuff = item.dwBuff;
```
Comment (obligaron) [inline Source/msg.cpp:2965]: We have this logic now in three places.
Could we add a helper method `SetItemForTCmdChItem` or something like this?

File: Source/msg.cpp [hunk 2]
```diff
@@ -2893,6 +2949,47 @@ void NetSendCmdDelItem(bool bHiPri, uint8_t bLoc)
 		NetSendLoPri(MyPlayerId, (byte *)&cmd, sizeof(cmd));
 }
 
+void NetSendCmdChInvItem(bool bHiPri, int invGridIndex)
+{
+	TCmdChItem cmd;
+
+	int8_t invListIndex = abs(MyPlayer->InvGrid[invGridIndex]) - 1;
+	Item &item = MyPlayer->InvList[invListIndex];
```
Comment (obligaron) [inline Source/msg.cpp:2957]:
```suggestion
	const Item &item = MyPlayer->InvList[invListIndex];
```

File: Source/msg.cpp [hunk 3]
```diff
@@ -2893,6 +2949,47 @@ void NetSendCmdDelItem(bool bHiPri, uint8_t bLoc)
...
+		NetSendHiPri(MyPlayerId, (byte *)&cmd, sizeof(cmd));
+	else
+		NetSendLoPri(MyPlayerId, (byte *)&cmd, sizeof(cmd));
+}
+
+void NetSendCmdChBeltItem(bool bHiPri, int beltIndex)
+{
+	TCmdChItem cmd;
+
+	Item &item = MyPlayer->SpdList[beltIndex];
```
Comment (obligaron) [inline Source/msg.cpp:2977]:
```suggestion
	const Item &item = MyPlayer->SpdList[beltIndex];
```

**Review by obligaron — APPROVED** [review]
lgtm, only some nits 🙂 
Thanks for working on this ❤️

File: Source/msg.cpp (hunk 1 above)
Comment (StephenCWills) [reply Source/msg.cpp:2965]: I'm hesitant to do that just yet because I will probably need to replace two of these places with `TCmdPItem` instead.

Comment (obligaron) [reply Source/msg.cpp:2965]: I'm fine with leaving them for now and refactor them later. 🙂

**Review by AJenbo — APPROVED** [review]

**Pull request merged**